src/cachemachine/types.py

```diff
@@ -102,6 +102,22 @@ def dump(self) -> List[Dict[str, str]]:
         return [asdict(i) for i in self]
 
 
+@dataclass
```

Comment:
I'm confused; is the second list all images or all images XORed with desired images?